zkevm-circuits/src/pi_circuit2.rs

```diff
@@ -1773,14 +3243,297 @@ mod pi_circuit_test {
         block.eth_block.mix_hash = Some(*OMMERS_HASH);
         block.eth_block.nonce = Some(H64::from([0, 0, 0, 0, 0, 0, 0, 0]));
         block.eth_block.base_fee_per_gas = Some(U256::from(0));
+        block.eth_block.withdrawals_root = Some(H256::zero());
+        block.context.history_hashes = vec![U256::zero(); 256];
+        block.context.history_hashes[255] =
+            U256::from_big_endian(block.eth_block.parent_hash.as_fixed_bytes());
 
         let public_data = PublicData::new(&block, prover, Default::default());
 
         let k = 17;
 
         assert_eq!(
-            run::<Fr, MAX_TXS, MAX_CALLDATA>(k, public_data, None),
+            run::<Fr, MAX_TXS, MAX_CALLDATA>(k, public_data, None, None),
             Ok(())
         );
     }
+
+    #[test]
+    fn test_blockhash_calc_short_values() {
+        const MAX_TXS: usize = 8;
+        const MAX_CALLDATA: usize = 200;
+        let prover =
+            Address::from_slice(&hex::decode("df08f82de32b8d460adbe8d72043e3a7e25a3b39").unwrap());
+
+        let mut block = witness::Block::<Fr>::default();
+        block.eth_block.parent_hash = *OMMERS_HASH;
+        block.eth_block.author = Some(prover);
+        block.eth_block.state_root = *OMMERS_HASH;
+        block.eth_block.transactions_root = *OMMERS_HASH;
+        block.eth_block.receipts_root = *OMMERS_HASH;
+        block.eth_block.logs_bloom = Some([0; LOGS_BLOOM_SIZE].into());
+        block.eth_block.extra_data = eth_types::Bytes::from([0; 0]);
+        block.eth_block.mix_hash = Some(*OMMERS_HASH);
+        block.eth_block.nonce = Some(H64::from([0, 0, 0, 0, 0, 0, 0, 0]));
+        block.context.number = U256::from(0x75);
+        block.context.gas_limit = 0x76;
+        block.eth_block.gas_used = U256::from(0x77);
+        block.context.timestamp = U256::from(0x78);
+        block.context.base_fee = U256::from(0x79);
+        block.context.difficulty = U256::from(0);
+        block.eth_block.withdrawals_root = Some(H256::from_slice(
+            &hex::decode("61223344dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49353")
+                .unwrap(),
+        ));
+        block.context.history_hashes = vec![U256::zero(); 256];
+        block.context.history_hashes[255] =
+            U256::from_big_endian(block.eth_block.parent_hash.as_fixed_bytes());
+
+        let public_data = PublicData::new(&block, prover, Default::default());
```

Comment:
Because each test needs to set up this data, will be a bit nicer if there's a function that builds a dummy block using some test values, and then per test you can just overwrite the values that you need for that specific test. Or something like that, just so there's less duplicated stuff and it's easier to see what data is being tested.

Reply:
makes sense - I've factored these out